Resources/template.json

```diff
@@ -544,6 +570,38 @@
         }
       }
     },
+    {
+      "name": "appGatewayNetworkSecurityGroup",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2017-05-10",
+      "dependsOn": [
+        "sharedManagedInstanceEnvironment"
+      ],
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[concat(variables('localTemplatesUrl'), 'network-security-groups/ag-nsg.json')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "name": {
+            "value": "[parameters('appGatewayNetworkSecurityGroup')]"
+          },
+          "agSubnetName": {
+            "value": "[parameters('appGatewaySubnetName')]"
+          },
+          "vnetName": {
+            "value": "[if(equals(parameters('envType'),'dev'), reference('sharedManagedInstanceEnvironment','2017-05-10').outputs.virtualNetworkName.value, parameters('appGatewayVnetName'))]"

```

Comment:
Should this condition just be testing for 'dev' or 'dev' & 'sit'?

Reply:
Just dev, the description for the param says "Type of environment, either dev or prod"